### PR TITLE
Issue175

### PR DESCRIPTION
**Pull request by kbaird**
Makes `FormatRequired` plug accept a legal multi-RIO payload.

### CODE REVIEW

File: lib/jsonapi.ex [hunk 1]
```diff
@@ -9,7 +9,7 @@ defmodule JSONAPI do
   in your `config/config.exs`:
       config :jsonapi, :json_library, Jason
   """
-  def json_library() do
+  def json_library do
```
Comment (kbaird) [inline lib/jsonapi.ex]: Everything in this file was to satisfy Credo.

File: lib/jsonapi/serializer.ex [hunk 2]
```diff
@@ -5,8 +5,8 @@ defmodule JSONAPI.Serializer do
 
   import JSONAPI.Ecto, only: [assoc_loaded?: 1]
 
-  alias JSONAPI.Utils.String, as: JString
-  alias JSONAPI.Config
+  alias JSONAPI.{Config, Utils}
+  alias Utils.String, as: JString
```
Comment (kbaird) [inline lib/jsonapi/serializer.ex]: This was also to satisfy Credo.

File: lib/jsonapi/plugs/format_required.ex [hunk 3]
```diff
@@ -5,12 +5,20 @@ defmodule JSONAPI.FormatRequired do
 
   import JSONAPI.ErrorView
 
+  # Cf. https://jsonapi.org/format/#crud-updating-to-many-relationships
+  @update_has_many_relationships_methods ~w[DELETE PATCH POST]
```
Comment (kbaird) [inline lib/jsonapi/plugs/format_required.ex]: Technically, DELETE is handled earlier by the first clause. But it is technically among the allowed methods for the multi-RIO payload, so I thought including it here made more sense than omitting it.

**Comment by jherdman** [comment]
The Credo changes are a bit surprising. On my dev machine, and CI, `mix credo` reports no concerns. Would you mind discussing your set up? Perhaps it'll help us figure out the discrepancies.

**Comment by kbaird** [comment]
Sure. 

```
$ iex -v
Erlang/OTP 21 [erts-10.2.3] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

IEx 1.8.1 (compiled with Erlang/OTP 20)
```

```
$ mix credo --strict
Ignoring an undefined check: Credo.Check.Warning.NameRedeclarationByAssignment
Ignoring an undefined check: Credo.Check.Warning.NameRedeclarationByCase
Ignoring an undefined check: Credo.Check.Warning.NameRedeclarationByDef
Ignoring an undefined check: Credo.Check.Warning.NameRedeclarationByFn
Checking 17 source files ...

info: the following checks were skipped because they're not compatible with
your version of Elixir (1.8.1).

   1) Credo.Check.Refactor.MapInto


  Code Readability                                                              
┃ 
┃ [R] ↘ The alias `JSONAPI.Config` is not alphabetically ordered among its 
┃       group.
┃       lib/jsonapi/serializer.ex:9:9 #(JSONAPI.Serializer)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/jsonapi.ex:33 #(JSONAPI.mime_type)
┃ [R] ↘ Do not use parentheses when defining a function which has no arguments.
┃       lib/jsonapi.ex:12 #(JSONAPI.json_library)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.4 seconds (0.01s to load, 0.4s running checks)
168 mods/funs, found 3 code readability issues.
```

**Comment by jherdman** [comment]
OH! Gotcha. OK, AFAIK we're not using the `--strict` flag. Would you mind reverting the Credo changes for now? If we want to move to strict that may be a worthy PR in-and-of-itself.

**Comment by kbaird** [comment]
Credo changes now in https://github.com/jeregrine/jsonapi/pull/177

**Comment by jherdman** [comment]
@kbaird could you rebase this against master? It has your Credo changes, and I want to make sure the build is still green (I'm sure it is, but ya know... safe vs. sorry).

**Comment by kbaird** [comment]
Rebased via the procedure at https://gist.github.com/ravibhure/a7e0918ff4937c9ea1c456698dcd58aa

**Comment by doomspork** [comment]
@kbaird something isn't quite right but I think it's c1b1849 that's the culprit, it adds `()` to the functions and fails Credo.

There's 2 ways we can fix this:

```shell
$ git fetch --all --prune
$ git rebase upstream/master -i --autostash --autosquash
# I do interactive rebase so I can squash, fixup, and rename commits.
# At the screen that pops up, delete the c1b1849 line, save and exit.
# That should remove that commit for us.
$ git push origin master -f
```

You could add a new commit that undoes those changes, when we squash it won't matter 😁 

I'd be happy to help you if that doesn't resolve the issues. I'd expect the `credo --strict` call to pass on your branch.

**Comment by kbaird** [comment]
@doomspork I just backed it up one commit. Let me know if it needs any more attention from me. Thanks.

**Comment by jherdman** [comment]
@doomspork you good with this? I think we're all set to merge.

**Comment by doomspork** [comment]
Thanks so much @kbaird and @jherdman 😁

**Pull request merged**